tests/basic.rs

```diff
@@ -37,6 +37,42 @@ fn get_key_value_empty() {
     }
 }
 
+#[test]
+fn replace_empty() {
+    let map = HashMap::<usize, usize>::new();
+
+    {
+        let guard = epoch::pin();
+        let old = map.replace_node(&42, None, None, &guard);
+        assert!(old.is_none());
+    }
+}
+
+#[test]
+fn replace_existing() {
+    let map = HashMap::<usize, usize>::new();
+    {
+        let guard = epoch::pin();
+        map.insert(42, 42, &guard);
+        let old = map.replace_node(&42, Some(10), None, &guard);
+        assert!(old.is_none());
+        assert_eq!(*map.get(&42, &guard).unwrap(), 10);
+    }
+}
+
+#[test]
+fn replace_old_value_non_matching() {
+    let map = HashMap::<usize, usize>::new();
+    {
+        let guard = epoch::pin();
+        map.insert(42, 42, &guard);
+        // TODO: fit test with an actual non valid old_value
+        let old = map.replace_node(&42, None, None, &guard);
```

Comment:
@jonhoo , How could I create a testing `Shared<V>` here?, Also, how can I actually retrieve that from the map?
Sorry for the mess, I'm just dealing with this crossbeam pointers for the first time.

Reply:
I think you can just do
```rust
let old_value = Shared::from(map.get(&42, &guard).unwrap());
```
Since `replace_node` is a private function though, this probably has to be a unit-test directly in `map.rs`.

Reply:
I cannot create the pointer like that. Maybe I'm using the wrong type?
Same issue [here](https://github.com/jonhoo/flurry/pull/36#discussion_r372006985)